sklearn/feature_extraction/text.py

```diff
@@ -1151,11 +1157,8 @@ def transform(self, X, copy=True):
         -------
         vectors : sparse matrix, [n_samples, n_features]
         """
-        if hasattr(X, 'dtype') and np.issubdtype(X.dtype, np.floating):
-            # preserve float family dtype
-            X = sp.csr_matrix(X, copy=copy)
-        else:
-            # convert counts or binary occurrences to floats
+        X = check_array(X, accept_sparse='csr', dtype=FLOAT_DTYPES)
```

Comment:
add `copy=copy` otherwise this will ignore the copy parameter passed to transform I think?

Reply:
right and I can remove from the `if sparse...`